### PR TITLE
value.ToStruct()

### PR DESCRIPTION
**Pull request by rsdoiel**
I added a convenience function to value called ToStruct. It wraps an value.Export() with a json.Marshal(), json.Unmarshal() call to populate a struct.  Here's how it might be used.

aStruct := {
   One: 1 `json:"one"`
   Two: "Two" `json:"two"`
}{}

val, _ := vm.Run(`(function(){ return { one: 1, two: "two"};}())`)
err := val.ToStruct(&aStruct)

fmt.Printf("One: %d, Two: %s\n", aStruct.One, aStruct.Two)

### CODE REVIEW

File: AUTHORS [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
```
Comment (darkliquid) [inline AUTHORS]: Is this file necessary?

File: value.go [hunk 2]
```diff
@@ -1023,3 +1024,29 @@ func stringToReflectValue(value string, kind reflect.Kind) (reflect.Value, error
...
 }
+
+// ToStruct will attempt populate a struct passed in as a parameter.
+//
+// ToStruct returns an error if it runs into a problem.
+//
+// Example:
+// a := struct{One int, Two string}{}
+// val, _ := vm.Run(`(function (){ return {One: 1, Two: "two"}}())`)
+// _ := val.ToSruct(&a)
```
Comment (darkliquid) [inline value.go]: typo in comment here

File: value.go [hunk 3]
```diff
@@ -1023,3 +1024,29 @@ func stringToReflectValue(value string, kind reflect.Kind) (reflect.Value, error
...
+// fmt.Printf("One: %d, Two: %s\n", a.One, a.Two)
+//
+func (value Value) ToStruct(aStruct interface{}) error {
+	raw, err := value.Export()
+	if err != nil {
+		return fmt.Errorf("failed to export value, %s", err)
+	}
+	src, err := json.Marshal(raw)
```
Comment (darkliquid) [inline value.go]: I dunno, it's seems kind of wrong to marshal into json only to immediate demarshal it - like there really should just be a direct marshalling to structs. However, obviously writing that is a lot more work so I can understand reusing the json package to do this, even if the intermediate step of storage as a byte array is an unnecessary annoyance.

**Comment by darkliquid** [comment]
Also, any chance you could combine all these commits into a single commit so the main commit tree doesn't get dirtied up too much?

**Comment by deoxxa** [comment]
Hrmmm, I'm not a fan of the intermediate JSON step. Maybe [mapstructure](https://github.com/mitchellh/mapstructure) could provide inspiration or code to do this without serialising/deserialising everything.

**Comment by rsdoiel** [comment]
Sorry to be out of touch for a bit. I wasn't happy about the marshal and unmashal step either it was just the shortest route to a problem I had to solve quickly.  Let me look at mapstructure and see what I can come up with then I'll submit a revised pull request. Thanks, Robert
